### PR TITLE
Betaversion

### PR DESCRIPTION
**Pull request by vitalygv**
Hello. Please check updated version.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    </div>
+
+    <div class="container"></div>
+
+    <script>
+        const container = document.querySelector('.container');
+
+        function createGrid(gridSize) {
+            let gridSizeSq = gridSize ** 2;
+            container.style.gridTemplateColumns = `repeat(${gridSize}, 1fr)`;
```
Comment (krabique) [inline index.html:23]: This looks interesting, what does it do?

File: index.html [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                })
+                container.appendChild(gridDiv);
+            }
+        }
+
+        function removeGrid() {
+            while (container.firstChild) {
+                container.removeChild(container.lastChild);
+            }
+        }
```
Comment (krabique) [inline index.html:38]: Does it remove all the elements from container? I wonder if there is a way to just remove them all at once, like set container content to blank or something.

File: index.html [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+        createGrid(16); //create basic grid
```
Comment (krabique) [inline index.html:40]: This method's name is pretty descriptive, I don't think you need a comment for that.

File: index.html [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+        const resetBtn = document.querySelector('#reset');
+
+        console.log(resetBtn);
```
Comment (krabique) [inline index.html:44]: It was probably used for debugging, no reason to keep it though

File: index.html [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        resetBtn.addEventListener("click", () => {
+            removeGrid();
+            createGrid(16);
+        });
+
+        const gridSizeBtn = document.querySelector('#gridSize');
+
+        gridSizeBtn.addEventListener('click', () => {
+            let gridInput = prompt('Enter grid size up to 100');
```
Comment (krabique) [inline index.html:53]: can be a const

File: index.html [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+            if (gridInput > 100) {
+                alert("Max grid size 100")
+            }
+            else {
+                removeGrid();
+                createGrid(gridInput);
+            }
+        })
```
Comment (krabique) [inline index.html:61]: This looks good, and we can improve it a little further. `removeGrid` and `createGrid` are used together in two places here, so they can be moved to a new function `resetGrid`. 

After you do that, you can then refactor this last anonymous function's code to use a ternary operator
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator

So this
```js
            if (gridInput > 100) {
                alert("Max grid size 100")
            }
            else {
                removeGrid();
                createGrid(gridInput);
            }
```
will look like this
```js
gridInput > 100 ? alert("Max grid size 100") : resetGrid(gridInput);
```

**Review by krabique — APPROVED** [review]

File: index.html (hunk 6 above)
Comment (krabique) [reply index.html:61]: And

```js
        resetBtn.addEventListener("click", () => {
            removeGrid();
            createGrid(16);
        });
```
will look like this
```js
resetBtn.addEventListener("click", () => resetGrid(16));
```

File: index.html (hunk 1 above)
Comment (krabique) [reply index.html:23]: Nevermind, it's grids afaik, I'm just not familiar with it

Comment (vitalygv) [reply index.html:23]: Yes, this creates grid. It's a property of Display: grid.

File: index.html (hunk 2 above)
Comment (vitalygv) [reply index.html:38]: Yes, it removes everything inside the parent container. That's the shortest solution I've found so far.

Comment (krabique) [reply index.html:38]: Yeah, it looks like the best solution there is

**Pull request merged**